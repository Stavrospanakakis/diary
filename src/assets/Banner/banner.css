html {
  scroll-behavior: smooth;
}

.banner {
  background-image:linear-gradient(rgba(0,0,0,0.3), rgba(0, 0, 0,0.3)) ,  url('banner.jpg') ;
  background-size: cover;
  background-position: center;
  height:80vh;
}

.banner .hero-area {
  width:100%;    
  height:100%;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  display: flex;
}

.banner .hero-area .hero-text {
  background: rgba(0, 0, 0, 0.3);
  border: 1px solid rgba(0, 0, 0, 0.3);
  border-radius: 1em;
  padding: 2em;
}