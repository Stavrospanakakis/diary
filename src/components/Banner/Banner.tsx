import React from 'react'
import { Link } from 'gatsby'

import '../../assets/Banner/banner.css'

const Banner: React.FC = () => {
	return (
		<div className="banner">
			<div className='hero-area'>
				<div className="hero-text ">
					<h1 className='text-xl sm:text-2xl md:text-6xl text-white text-center flex justify-center align-center'>I am Stavros Panakakis</h1>
					<h1 className='text-md sm:text-lg md:text-2xl text-white text-center flex justify-center align-center'>Welcome to my personal diary</h1>
					<Link className='flex justify-center align-center mt-5' to='#posts'>
						<button className='text-xs sm:text-sm md:text-lg text-white font-semibold hover:bg-purple-500 py-2 px-2 border border-purple-500 hover:border-transparent rounded'>
							Latest Posts
						</button>
					</Link>
				</div>
			</div>
		</div>

	)
}

export default Banner