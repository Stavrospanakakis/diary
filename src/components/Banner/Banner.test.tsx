import React from 'react'

import { render } from '@testing-library/react'

import Banner from './Banner'

describe('BAnner', () => {
	it('should render', () => {
		const { container } = render(<Banner />)
		expect(container).toMatchSnapshot()
	})
})