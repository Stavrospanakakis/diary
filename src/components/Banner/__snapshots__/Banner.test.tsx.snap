// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BAnner should render 1`] = `
<div>
  <div
    class="banner"
  >
    <div
      class="hero-area"
    >
      <div
        class="hero-text "
      >
        <h1
          class="text-xl sm:text-2xl md:text-6xl text-white text-center flex justify-center align-center"
        >
          I am Stavros Panakakis
        </h1>
        <h1
          class="text-md sm:text-lg md:text-2xl text-white text-center flex justify-center align-center"
        >
          Welcome to my personal diary
        </h1>
        <a
          class="flex justify-center align-center mt-5"
          href="/#posts"
        >
          <button
            class="text-xs sm:text-sm md:text-lg text-white font-semibold hover:bg-purple-500 py-2 px-2 border border-purple-500 hover:border-transparent rounded"
          >
            Latest Posts
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
